aep/general/0140/aep.md.j2

```diff
@@ -1,5 +1,165 @@
 # Field names
 
-**Note:** This AEP has not yet been adopted. See
-[this GitHub issue](https://github.com/aep-dev/aep.dev/issues/21) for more
-information.
+Naming fields in a way that is intuitive to users can often be one of the most
+challenging aspects of designing an API. This is true for many reasons; often a
+field name that seems entirely intuitive to the author can baffle a reader.
+
+Additionally, users rarely use only one API; they use many APIs together. As a
+result, a single company using the same name to mean different things (or
+different names to mean the same thing) can often cause unnecessary confusion,
+because users can no longer take what they've already learned from one API and
+apply that to another.
+
+In short, APIs are easiest to understand when field names are simple,
+intuitive, and consistent with one another.
+
+## Guidance
+
+Field names **should** be in correct American English.
+
+Field names **should** clearly and precisely communicate the concept being
+presented and avoid overly general names that are ambiguous. That said, field
+names **should** avoid including unnecessary words. In particular, avoid
+including adjectives that always apply and add little cognitive value. For
+example, a `proxy_settings` field might be as helpful as
+`shared_proxy_settings` if there is no unshared variant.
+
+**Important:** Field names often appear in generated client surfaces. Ensure
+they are appropriately descriptive and of suitable length.
+
+### Case
+
+{% tab proto %} Field definitions in protobuf files **must** use
+`lower_snake_case` names. These names are mapped to an appropriate naming
+convention in JSON and in generated code.
+
+{% tab oas %}
+
+Field definitions in OAS schemas **must** use `camelCase` names.
+
+{% endtabs %}
+
+**Note:** Examples throughout this AEP use `snake_case`. This should not be
+taken to mean that this guidance is protobuf-specific. Any guidance not in a
+protobuf tab applies to all APIs, with `snake_case` examples converted to
+`camelCase` for OAS APIs.
+
+Each word in a field name **must not** begin with a number, because it creates
+ambiguity when converting between snake case and camel case. Similarly, fields
+**must not** contain leading, trailing, or adjacent underscores.
+
+### Uniformity
+
+APIs **should** endeavor to use the same name for the same concept and
+different names for different concepts wherever possible. This includes names
+across multiple APIs, in particular if those APIs are likely to be used
+together.
+
+### Arrays / Repeated fields
+
+Arrays (in OAS) and repeated fields (in protobuf) **must** use the proper
+plural form, such as `books` or `authors`. On the other hand, singular fields
+**should** use the singular form such as `book` or `author`.
```

Comment:
Upgrade to must?

(A singular field called "books" feels unintuitive to me)

Reply:
Done.